module/zfs/spa.c

```diff
@@ -7173,6 +7173,16 @@ spa_vdev_add(spa_t *spa, nvlist_t *nvroot)
 		}
 	}
 
+	if (ashift_check) {
+		for (int c = 0; c < vd->vdev_children; c++) {
+			tvd = vd->vdev_child[c];
+			if (tvd->vdev_ashift != spa->spa_max_ashift) {
```

Comment:
What happens if you start with an pool with ashift 9, then force add a vdev with ashift 12, then attempt to add a vdev with ashift 9? It looks like we will fail the attempt because `spa_max_ashift` will become 12, right? Is this the behavior we want? Perhaps it would be better if we ignore the ashift check once we have a "mixed" pool?